pipeline/pipelines/code_generation_pipeline.py

```diff
@@ -105,8 +105,8 @@ def do_build_flow(self, **kwargs):
     def validate_kwargs(self, **kwargs):
         req_args = _VGEN_REQUIRED[:]
         if 'publish_env' in kwargs:
-            req_args += _PYTHON_PUB_REQUIRED
-            _validate_publish_kwargs(**kwargs)
+            req_args += _PUB_REQUIRED
+            _validate_publish_env(_PYTHON_PUB_ENVS, kwargs['publish_env'])
```

Comment:
Can we remove the repeating similar code again here?
Ideally it would be:
```
def validate_kwargs(self, **kwargs):
  req_args = _VGEN_REQUIRED[:]
  _validate_publish_env(req_args, _PYTHON_PUB_ENVS, **kwargs)
  _validate_codegen_kwargs(req_args, **kwargs)
```

Please consider moving more vode into _validate_publish_env to achieve this. This would simplify the code.